src/test/java/com/googlecode/jsonrpc4j/server/JsonRpcServerAnnotateMethodVarArgsTest.java

```diff
@@ -0,0 +1,114 @@
+package com.googlecode.jsonrpc4j.server;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.googlecode.jsonrpc4j.JsonRpcBasicServer;
+import com.googlecode.jsonrpc4j.JsonRpcMethod;
+import com.googlecode.jsonrpc4j.VarArgsUtil;
+import org.easymock.EasyMockRunner;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.googlecode.jsonrpc4j.JsonRpcBasicServer.RESULT;
+import static com.googlecode.jsonrpc4j.util.Util.*;
+
+@RunWith(EasyMockRunner.class)
+public class JsonRpcServerAnnotateMethodVarArgsTest {
+
+	private ServiceWithVarArgsImpl mockService = new ServiceWithVarArgsImpl();
+	
+	private ByteArrayOutputStream byteArrayOutputStream;
+	private JsonRpcBasicServer jsonRpcServerAnnotatedMethod;
+	
+	@Before
+	public void setup() {
+		byteArrayOutputStream = new ByteArrayOutputStream();
+		jsonRpcServerAnnotatedMethod = new JsonRpcBasicServer(mapper, mockService, ServiceWithVarArgsImpl.class);
+	}
+
+	@Test
+	public void callMethodWithVarArgParameters() throws Exception {
+	    Map<String, Object> paramsMap = new HashMap<>();
+        paramsMap.put("argOne","one");
+        paramsMap.put("argTwo",2);
+        paramsMap.put("argThree","three");
+        paramsMap.put("argFour", 4);
+        paramsMap.put("argFive", (Object)"five");
+        paramsMap.put("argSix", 6.0f);
+        paramsMap.put("argSeven", 7d);
+        Object[] callParams = new Object[paramsMap.size()*2];
+        int i=0;
+        for (Map.Entry<String, Object> entry : paramsMap.entrySet()) {
+            callParams[i++] = entry.getKey();
+            callParams[i++] = entry.getValue();
+        }
+
+        jsonRpcServerAnnotatedMethod.handleRequest(
```

Comment:
again condensed it, in the age of widescreen no need to stop at 80 chars, the project uses 120 chars